tests/network/hotplug.go

```diff
@@ -405,6 +405,144 @@ var _ = SIGDescribe("nic-hotunplug", func() {
+			var err error
+			Eventually(func() v1.VirtualMachineInstancePhase {
+				newVMI, err = kubevirt.Client().VirtualMachineInstance(vm.Namespace).Get(context.Background(), vm.Name, &metav1.GetOptions{})
+				if err != nil || vmi.UID == newVMI.UID {
+					return v1.VmPhaseUnset
+				}
+				return newVMI.Status.Phase
+			}, 90*time.Second, 1*time.Second).Should(Equal(v1.Running))
+			libwait.WaitUntilVMIReady(newVMI, console.LoginToAlpine)
+
+			By("verify unplugged interface does not exist in the new VMI spec")
+			newVMI = verifyDynamicInterfaceChange(newVMI, plugMethod)
+			Expect(libnet.InterfaceExists(newVMI, guestNICName)).ToNot(Succeed(),
+				"unplugged interface should not exist in guest after VM restart")
+		},
+			Entry("In place", decorators.InPlaceHotplugNICs, inPlace),
+			Entry("Migration based", decorators.MigrationBasedHotplugNICs, migrationBased),
+		)
+	})
+
+	Context("a stopped VM", func() {
+		var vm *v1.VirtualMachine
+		var vmi *v1.VirtualMachineInstance
+
+		BeforeEach(func() {
+			By("create stopped VM")
+			opts := append(
+				libvmi.WithMasqueradeNetworking(),
+				libvmi.WithNetwork(libvmi.MultusNetwork(linuxBridgeNetworkName1, networkName)),
+				libvmi.WithNetwork(libvmi.MultusNetwork(linuxBridgeNetworkName2, networkName)),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(linuxBridgeNetworkName1)),
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithBridgeBinding(linuxBridgeNetworkName2)),
+			)
+			vmi = libvmi.NewAlpineWithTestTooling(opts...)
+			vm = tests.NewRandomVirtualMachine(vmi, false)
+
+			var err error
+			vm, err = kubevirt.Client().VirtualMachine(util.NamespaceTestDefault).Create(context.Background(), vm)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		It("cannot be subject to **hot** unplug, but will mutate the template.Spec on behalf of the user", func() {
+			removeIfaceOpts := &v1.RemoveInterfaceOptions{
+				Name: linuxBridgeNetworkName2,
+			}
+			Expect(kubevirt.Client().VirtualMachine(vm.Namespace).RemoveInterface(context.Background(), vm.Name, removeIfaceOpts)).To(Succeed())
+
+			By("wait for the requested unplug interface spec update and have 'absent' state")
+			Eventually(func() []v1.Interface {
+				var err error
+				vm, err = kubevirt.Client().VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, &metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred())
+				return vm.Spec.Template.Spec.Domain.Devices.Interfaces
+			}, 30*time.Second).Should(ConsistOf(
+				*v1.DefaultMasqueradeNetworkInterface(),
+				v1.Interface{
+					Name: linuxBridgeNetworkName1,
+					InterfaceBindingMethod: v1.InterfaceBindingMethod{
+						Bridge: &v1.InterfaceBridge{},
+					},
+				},
+				v1.Interface{
+					Name:  linuxBridgeNetworkName2,
+					State: v1.InterfaceStateAbsent,
+					InterfaceBindingMethod: v1.InterfaceBindingMethod{
+						Bridge: &v1.InterfaceBridge{},
+					},
+				},
+			))
+
+			Expect(vm.Spec.Template.Spec.Networks).To(ConsistOf(
```

Comment:
A lot of noise IMO.

Just save the VM spec from before the interface removal command, then show that the only difference from the previous and new is the specific interface having the state changed to absent.

Reply:
Done